cmd/zfs/zfs_iter.c

```diff
@@ -134,13 +134,23 @@ zfs_callback(zfs_handle_t *zhp, void *data)
 	    ((cb->cb_flags & ZFS_ITER_DEPTH_LIMIT) == 0 ||
 	    cb->cb_depth < cb->cb_depth_limit)) {
 		cb->cb_depth++;
-		if (zfs_get_type(zhp) == ZFS_TYPE_FILESYSTEM)
+
+		/*
+		 * If we are not looking for filesystems, we don't need to
+		 * recurse into filesystems when we are at our depth limit.
+		 */
+		if ((cb->cb_depth < cb->cb_depth_limit ||
+		    (cb->cb_types & ZFS_TYPE_FILESYSTEM)) &&
+		    zfs_get_type(zhp) == ZFS_TYPE_FILESYSTEM)
```

Comment:
Should this `if` have braces? The second part of the diff exists only to add them to the `if` below.

Reply:
Yes indeed it should, @tcaputi can you please add these.